### PR TITLE
Add more detailed results to context benchmark

### PR DESCRIPTION
**Pull request by janhartman**
This PR introduces the usage of `GetCodyContextAlternatives` endpoint to fetch more detailed context results for improved evaluations. See https://github.com/sourcegraph/cody-leaderboard/pull/19/ for the eval part - both PRs need to be merged at the same time to avoid breakage.

## Test plan
Tested with https://github.com/sourcegraph/cody-leaderboard/pull/19/

### CODE REVIEW

File: agent/src/cli/command-bench/strategy-chat-context-types.ts [hunk 1]
```diff
@@ -178,18 +173,8 @@ export async function readExamplesFromCSV(filePath: string): Promise<{
     }
 }
 
-/**
- * Note: this mutates evalOutput to remove the content field from actualContext context items.
- */
 export async function writeYAMLMetadata(outputFile: string, evalOutput: EvalOutput): Promise<void> {
```
Comment (janhartman) [inline agent/src/cli/command-bench/strategy-chat-context-types.ts:176]: We don't need the YAML output to also contain all results, we have the CSV file for that.

File: agent/src/cli/command-bench/strategy-chat-context-types.ts [hunk 2]
```diff
@@ -258,5 +236,5 @@ export function contextItemFromString(item: string): EvalContextItem {
 }
 
 export function contextItemToString(item: EvalContextItem): string {
-    return `${item.repoName}:${item.path}:${item.startLine}-${item.endLine}`
+    return `${item.retriever}:${item.repoName}:${item.path}:${item.startLine}-${item.endLine}`
```
Comment (janhartman) [inline agent/src/cli/command-bench/strategy-chat-context-types.ts:239]: A bit torn between this and just serializing it as JSON.

File: agent/src/cli/command-bench/strategy-chat-context.ts [hunk 3]
```diff
@@ -54,8 +54,9 @@ export async function evaluateChatContextStrategy(
         ? siteVersion.match(/-([0-9a-f]{7,40})$/)?.[1]
         : siteVersion
     const currentTimestamp = new Date().toISOString()
+    const date = currentTimestamp.split('T')[0]
 
-    const outputBase = `${inputBasename}__${shortSiteVersion}`
+    const outputBase = `${inputBasename}__${date}__${shortSiteVersion}`
```
Comment (janhartman) [inline agent/src/cli/command-bench/strategy-chat-context.ts:59]: Outputs are now also denoted with dates which isn't strictly necessary but it is helpful.

File: agent/src/cli/command-bench/strategy-chat-context-types.ts [hunk 4]
```diff
@@ -116,15 +118,8 @@ function exampleFromCsvRecord(record: any): Example {
     }
 }
 
-interface Stats {
-    essentialRecall5: number
-    essentialRecall10: number
-    essentialRecall: number
-}
```
Comment (janhartman) [inline agent/src/cli/command-bench/strategy-chat-context-types.ts:123]: All metrics are now computed inside the leaderboard, there's no need for this to be here.

File: agent/src/cli/command-context/command-context.ts [hunk 5]
```diff
@@ -1,279 +0,0 @@
-import * as fs from 'node:fs/promises'
```
Comment (janhartman) [inline agent/src/cli/command-context/command-context.ts:1]: This was unused

File: agent/src/cli/command-bench/strategy-chat-context-types.ts (hunk 2 above)
Comment (AyushmunotSG18) [reply agent/src/cli/command-bench/strategy-chat-context-types.ts:239]: Hey @janhartman,
I have made changes to in command-bench in this [PR](https://github.com/sourcegraph/cody/pull/6087), to ensure that they will be running on the modified clickable link format and the this current format. Can you approve that PR, and the merge those changes to this?

**Review by beyang — APPROVED** [review]

**Review by AyushmunotSG18 — APPROVED** [review]

**Pull request merged**